Makefile

```diff
@@ -2,43 +2,87 @@ SHELL = /bin/bash
+face_recognition = $(docker_repo)/faceanalysis_facerecognition:$(build_tag)
+faceapi = $(docker_repo)/faceanalysis_faceapi:$(build_tag)
+facenet = $(docker_repo)/faceanalysis_facenet:$(build_tag)
+insightface = $(docker_repo)/faceanalysis_insightface:$(build_tag)
 
+get_famous_people_list = $(docker_repo)/faceanalysis_getfamouspeoplelist:$(build_tag)
+get_famous_people_photos = $(docker_repo)/faceanalysis_getfamouspeoplephotos:$(build_tag)
+preprocessor = $(docker_repo)/faceanalysis_preprocessor:$(build_tag)
+
+.PHONY: build-dev
 build-dev:
 	DOCKER_REPO="$(docker_repo)" BUILD_TAG="$(build_tag)" DEVTOOLS="true" \
     docker-compose build
 
+.PHONY: build-prod
 build-prod:
 	DOCKER_REPO="$(docker_repo)" BUILD_TAG="$(build_tag)" \
     docker-compose build
 
+.PHONY: build-scripts
+build-scripts:
+	docker build -t "$(preprocessor)" ./scripts/preprocessor
+	docker build -t "$(get_famous_people_list)" ./scripts/get_famous_people_list
+	docker build -t "$(get_famous_people_photos)" ./scripts/get_famous_people_photos
+
+.PHONY: build-algorithms
 build-algorithms:
-	docker build -t "$(docker_repo)/faceanalysis_facerecognition:$(build_tag)" algorithms/face_recognition
-	docker build -t "$(docker_repo)/faceanalysis_faceapi:$(build_tag)" algorithms/FaceApi
-	docker build -t "$(docker_repo)/faceanalysis_facenet:$(build_tag)" algorithms/facenet
-	docker build -t "$(docker_repo)/faceanalysis_insightface:$(build_tag)" algorithms/insightface
+	docker build -t "$(face_recognition)" algorithms/face_recognition
+	docker build -t "$(faceapi)" algorithms/FaceApi
+	docker build -t "$(facenet)" algorithms/facenet
+	docker build -t "$(insightface)" algorithms/insightface
 
+.PHONY: release-server
 release-server: build-prod
 	DOCKER_REPO="$(docker_repo)" BUILD_TAG="$(build_tag)" \
     docker-compose push
 
+.PHONY: release-algorithms
 release-algorithms: build-algorithms
-	docker push "$(docker_repo)/faceanalysis_facerecognition:$(build_tag)"
-	docker push "$(docker_repo)/faceanalysis_faceapi:$(build_tag)"
-	docker push "$(docker_repo)/faceanalysis_facenet:$(build_tag)"
-	docker push "$(docker_repo)/faceanalysis_insightface:$(build_tag)"
+	docker push "$(face_recognition)"
+	docker push "$(faceapi)"
+	docker push "$(facenet)"
+	docker push "$(insightface)"
+
+.PHONY: pylint-scripts
+pylint-scripts: build-scripts
+	docker run -v $$PWD/app/.pylintrc:/app/.pylintrc --entrypoint=sh "$(get_famous_people_list)" -c "pip -qqq install pylint && pylint --rcfile=/app/.pylintrc *.py"
+	docker run -v $$PWD/app/.pylintrc:/app/.pylintrc --entrypoint=sh "$(get_famous_people_photos)" -c "pip -qqq install pylint && pylint --rcfile=/app/.pylintrc *.py"
+	docker run -v $$PWD/app/.pylintrc:/app/.pylintrc --entrypoint=sh "$(preprocessor)" -c "pip -qqq install pylint && pylint --rcfile=/app/.pylintrc *.py"
+
+.PHONY: flake8-scripts
+flake8-scripts: build-scripts
+	docker run --entrypoint=sh "$(get_famous_people_list)" -c "pip -qqq install flake8 && flake8 *.py"
```

Comment:
With -qqq, is it advisable to silence critical level logging warnings?

Reply:
That should be fine. `pip install` is not expected to fail and if it does, the entire command will fail (due to the `&&`) so we know what to look into. Having low verbosity definitely is nicer in the CI run since otherwise the pip logs just drown out the lint messages.

Reply:
Reduced to `-qq` since that's also still non-verbose. Thanks for the pointer.